modules/standard/Spawn.chpl

```diff
@@ -711,60 +721,99 @@ module Spawn {
         to its output while the parent process is waiting for it to complete
         (and not consuming its output).
 
-    :arg error: optional argument to capture any error encountered
-                when waiting for the child process.
     :arg buffer: if `true`, buffer input and output pipes (see above).
-
    */
-  proc subprocess.wait(out error:syserr, buffer=true) {
+  proc subprocess.wait(buffer=true) throws {
+    try _throw_on_launch_error();
 
     if buffer {
-      this.communicate(error);
+      try this.communicate();
       return;
     }
 
-    error = this.spawn_error;
-    if !running then return;
+    if !running {
+      if this.spawn_error then
+        try ioerror(this.spawn_error, "in subprocess.wait");
+    }
 
-    on home {
+    // TODO: rethink this approach, perhaps with an Error list, or var Error
+    // - complete all these throwing calls, even if they throw
+    //   - extract a syserr if a SystemError is caught
+    // - at the end, throw an error if err is set
```

Comment:
I don't see this approach as so bad - it seems fine to me. So I don't really see why the above comment needs to be here.